ledger/simulation/simulation_eval_test.go

```diff
@@ -2190,3 +2226,387 @@ func TestMockTracerScenarios(t *testing.T) {
+						Version:   simulation.ResultLatestVersion,
+						LastRound: env.TxnInfo.LatestRound(),
+						TxnGroups: []simulation.TxnGroupResult{
+							{
+								Txns: []simulation.TxnResult{
+									{
+										Txn: transactions.SignedTxnWithAD{
+											ApplyData: transactions.ApplyData{
+												EvalDelta: expectedEvalDelta,
+											},
+										},
+										AppBudgetConsumed: ignoreAppBudgetConsumed,
+									},
+								},
+								FailedAt:          expectedFailedAt,
+								AppBudgetAdded:    700,
+								AppBudgetConsumed: ignoreAppBudgetConsumed,
+							},
+						},
+						EvalOverrides: simulation.ResultEvalOverrides{
+							AllowUnlimitedResourceAccess: true,
+						},
+					},
+				}
+			})
+		})
+	}
+}
+
+// TestUnlimitedResourceAccessBoxReadBudget tests app call behavior when exceeding box read budget
+// if AllowUnlimitedResourceAccess is enabled.
+func TestUnlimitedResourceAccessBoxReadBudget(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+	// Start with 8, since that's when boxes were introduced
+	for v := 8; v <= logic.LogicVersion; v++ {
+		v := v
+		t.Run(fmt.Sprintf("v%d", v), func(t *testing.T) {
+			t.Parallel()
+			simulationTest(t, func(env simulationtesting.Environment) simulationTestCase {
+				sender := env.Accounts[0]
+
+				program := fmt.Sprintf(`#pragma version %d
+txn ApplicationID
+bz end // Do nothing during create
+
+txn ApplicationArgs 0
+byte "create" // Create a box
+==
+bz end
+txn ApplicationArgs 1
+txn ApplicationArgs 2
+btoi
+box_create
+pop
+
+end:
+int 1
+`, v)
+
+				testAppID := env.CreateApp(sender.Addr, simulationtesting.AppParams{
+					ApprovalProgram:   program,
+					ClearStateProgram: fmt.Sprintf("#pragma version %d\n int 1", v),
+					LocalStateSchema: basics.StateSchema{
+						NumByteSlice: 1,
+					},
+				})
+
+				// Fund the app to cover min balance increase
+				env.TransferAlgos(sender.Addr, testAppID.Address(), 1_000_000)
+
+				bytesPerBoxReference := env.TxnInfo.CurrentProtocolParams().BytesPerBoxReference
+
+				// Create a box which exceeds the BytesPerBoxReference. This is ok because we supply
+				// another box ref during the call.
+				env.Txn(env.TxnInfo.NewTxn(txntest.Txn{
+					Type:          protocol.ApplicationCallTx,
+					Sender:        sender.Addr,
+					ApplicationID: testAppID,
+					ApplicationArgs: [][]byte{
+						[]byte("create"), []byte("A"), uint64ToBytes(2 * bytesPerBoxReference),
+					},
+					Boxes: []transactions.BoxRef{
+						{Name: []byte("A")},
+						{},
+					},
+				}).SignedTxn())
+
+				// Call the program with only the box reference whose size exceed
+				// BytesPerBoxReference. This would normally not be allowed.
+				txn := env.TxnInfo.NewTxn(txntest.Txn{
+					Type:            protocol.ApplicationCallTx,
+					Sender:          sender.Addr,
+					ApplicationID:   testAppID,
+					ApplicationArgs: [][]byte{{}},
+					Boxes: []transactions.BoxRef{
+						{Name: []byte("A")},
```

Comment:
Yes, normally you would need at least 2 box refs to satisfy the read budget for this transaction.

You could argue that raising box read limits isn't necessary PR, since you could just omit all box refs from the transaction and access whatever you want during the program anyway, but I thought it seemed inconsistent to raise an error for this.

Reply:
yes, I figured that if you want unlimited resources, we better specify nothing in txn and let contract run. I am playing locally around it and removing all box refs seems fine in running.